hellomod.c

```diff
@@ -0,0 +1,17 @@
+#define MODULE
```

Comment:
Do you understand what the first 3 lines mean?

Reply:
Defines can alter the behavior of the code by emitting or adding lines at the preprocessing phase.
for example at linux\init.h the exit behaviour is different for modules.

Reply:
I mean why do you need to define `MODULE`, `LINUX` and `__KERNEL__`?

Reply:
__KERNEL__ - in header files which are used both by userspace and by kernel, __KERNEL__ is used to hide some kernel declarations so they won't be seen by userspace code.
MODULE - used for dynamically loaded modules (as opposed to built-in drivers), to declare different entry and exit points.
LINUX - I grepped the entire src code but found no if/ndefs that mention LINUX so I guess it has no code related functionallity.